garrysmod/gamemodes/sandbox/gamemode/editor_player.lua

```diff
@@ -115,6 +115,33 @@ list.Set( "DesktopWindows", "PlayerEditor", {
 
 		local bgtab = sheet:AddSheet( "#smwidget.bodygroups", bdcontrols, "icon16/cog.png" )
 
+		local animPanel = window:Add( "DPanel" )
+		animPanel:DockPadding( 8, 8, 8, 8 )
+
+		local animcontrolspanel = animPanel:Add( "DScrollPanel" )
+		animcontrolspanel:Dock( FILL )
+
+		local animsTable = {
+			"idle_all_scared",
+			"menu_gman",
+			"menu_zombie_01",
+			"menu_combine",
+			"pose_standing_02",
+		}
+
+		for v, n in pairs( animsTable ) do

```

Comment:
You can use ipairs here to be more efficient.

Reply:
> You can use ipairs here to be more efficient.

Thank you

Reply:
You're wrong @FlorianLeChat 
It's not more efficent here, as ipairs is only faster when JIT-compiled, here is not going to be JIT-compiled
On literally each line there is a NYI(closure)/C api call.
When not JIT-compiled, `ipairs` is 3.29 times slower than `pairs`


It is WRONG to use pairs, but not slower.

Reply:
Good point. I didn't notice that.